src/Spice86.Core/Emulator/Function/FunctionCatalogue.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace Spice86.Core.Emulator.Function;
+
+using Spice86.Shared.Emulator.Memory;
+
+public class FunctionCatalogue {
+    public FunctionCatalogue() : this(new Dictionary<SegmentedAddress, FunctionInformation>()) {
```

Comment:
So the same function can be in this dictionary with multiple SegmentedAddresses. Is that the way it should work or could that be problematic?

Reply:
That would be bad, would not make any sense. I guess you are thinking this could happen because we give a dictionary instead of a list? I will try to change it, this is an improvement

Reply:
I think I remember krondor having a function to convert a 32bit address to a seg:ofs combination. Probably because it loads large files and it's easier to calculate offsets that way.
Not sure if it also uses these to call functions, or only to access structs.

It also has a lot of indirect calls. Where the main code calls and address that has a jmp to the real function, wherever it is. But the library calls those functions directly. So functionally the same function, but with different addresses.

Reply:
If a function is called via a different segmented address, this implies wild things. Offsets calculations for near jumps could end up at different places like they do in ghidra, and code that accesses variables at CS:offset would access different values...